src/services/signer_service.ts

```diff
@@ -173,12 +200,91 @@ export class SignerService {
                     shouldValidate: false,
                 },
             );
-
+        await this._updateTransactionEntityToSubmittedAsync(txHash);
         return {
             ethereumTransactionHash,
             signedEthereumTransaction,
         };
     }
+    private async _watchForStuckTransactionsAsync(): Promise<void> {
+        while (true) {
```

Comment:
Thinking about this again, there is not any way to spin this down. That might not matter in production, but in testing it could be a headache. We should use `@0x/utils`'s `intervalUtils.setAsyncExcludingInterval` method. It has a similar effect to the existing code by disallowing the logic in the loop from executing multiple times in parallel. It is different in that, if it is still executing, it simply skips and waits `STUCK_TX_POLLING_INTERVAL_MS` time again.

The benefit is we get back a timer which can be cancelled on test tear down.